### DISCUSSION TITLE
There's a compilation problem

### DISCUSSION THREAD

**Discussion by codehooni** [goatshriek/stumpless#417] [Q&A]
Hello! I'm using various examples from the document to help with stumpless translation. I used the same example(docs/examples/journald/journald_example.c) used in the journal, but I got an error while compiling.

<img width="829" alt="error" src="https://github.com/goatshriek/stumpless/assets/101172414/03dd055f-be66-4d90-971a-663679cef338">

So I put the code as below and recompiled it. But another error came up.
```c
#include "stumpless/include/stumpless/target/journald.h"
#include "stumpless/src/target/journald.c"
``` 

<img width="749" alt="Screenshot 2024-04-21 at 10 00 44 AM" src="https://github.com/goatshriek/stumpless/assets/101172414/211e831b-7728-49ca-8e43-f3122f9a865f">

Is there anything I did wrong in the compilation process or is there something wrong with the journal code?

**Comment by goatshriek** [comment]
Have you configured and built the library with cmake and a build command (like make)? Also, the include issue could likely be resolved by passing two `-Idir`  parameters to gcc, where one dir is the path to the include folder of the repo, and the other is the path to the include folder of the configure directory.

EDIT: You could also just install the library, which should put the header files into a folder on your system that is in the include search path, such as `/usr/include`.

I don't think there is a lot of documentation on building the examples outside of the cmake targets to run them, so perhaps that is an area that could be improved and help you in this case. I will take a closer look at that when I have more time.

**Reply by codehooni** [reply]
I finished organizing and building the library using cmake and make! Other examples and basic logs were compiled and printed out normally.
As you said, I tried using -Idir, but it didn't compile properly. Did I type the wrong command? Here's the command I entered!
Thank you for your quick reply! Have a nice day :)
```
gcc test.c -o test -lstumpless -Istumpless/include/stumpless/target -Istumpless/src/target
```

**Reply by goatshriek** [reply]
Here are the commands I used to build a standalone source file from the build directory, without the library installed:

```sh
$ gcc test.c -lstumpless -L. -I include/ -I $CODEDIR/stumpless/include -o test
$ LD_LIBRARY_PATH=. ./test
```

One other possible thing that could be happening is that journald support was not included in the build due to a missing dependency. Check to see if `STUMPLESS_JOURNALD_TARGETS_SUPPORTED` is defined in `stumpless/config.h`. If it isn't, check the output from cmake during the build to see what library was missing for journald support (probably `libsystemd`). In that case if you want journald support, install the dependency and re-configure and build so that journald support is enabled.
